pkg/certificate/manager.go

```diff
@@ -336,3 +334,15 @@ func (m *Manager) ListIssuedCertificates() []*Certificate {
 	})
 	return certs
 }
+
+// SubscribeRotations returns a channel that outputs every certificate that is rotated by the manager.
+// The caller must call the returned method to close the channel.
+// WARNING: you cannot call wait on the returned channel on the same go routine you are issuing a certificate on.
+func (m *Manager) SubscribeRotations(key string) (chan interface{}, func()) {
+	ch := m.pubsub.Sub(key)
+	return ch, func() {
+		go m.pubsub.Unsub(ch)
+		for range ch {
```

Comment:
Why drain the channel here?

Reply:
nope, i was wrong. This is required due to the head of line blocking issue https://github.com/openservicemesh/osm/issues/4847.

If the channel is not empty, then the unsub command will never even occur